Exploration Shotgun

comma opened the pull request:
Daily meme from Uncle Chinsky

Added an exploration shotgun. It's in a wall locker on Charon, along with ammo, replacing netgun. What can possibly go wrong?
It can fire freely off-ship, but onboard you gotta register and get XO or CO to authorize it. Yes it's pain, yes it's intended.
It is only guaranteed to work with types of ammo it ships with - beanbags, flash, net rounds. Loading combat ammo voids warranty and may result in fun.

Also makes stun ammo more effective (or at all effective) against simple animals, and adds some actual stun. Overall, it's expected to waste 1 full shotshell holder per big beastie.

Also adds ammo box sprite cause why not

![](https://i.imgur.com/fM4iqUi.png)

![](https://i.imgur.com/zgIdOee.png)


![](https://i.imgur.com/5F5gZ27.png)

comma committed:
Adds exploration shotgun

It's a pump shotgun with few twists.
1.It's secure gun, meaning it needs to be registered and authorized. It starts unathorized to fire at all, and needs XO level access to be authorized. All in all, not intended to be fired shipside unless feces hit the ventilation.
2.When outside of Torch zlevels, secure chip is bypassed, letting you shoot your colleagues on away missions.
3.It's intended to fire beanbags, net shells and flares. If you try to fire something stronger, it will explode VIOLENTLY. Not a 100% chance, so you can try if you feel lucky.

As a side thing adds markings to shot holders because holy confusing batman. Also ammo box sprite.

Made simple animals actually be hurt by stun weapons, and made them bit stronger agains animals. Stun projectiles mostly use agony instead of straight up damage, so it was more or less no-effect on animals. Made it also stun animals for few ticks.

comma committed:
Map & Changelog

Glloyd commented:
That ammobox sprite is... Not the greatest. It doesn't look like anything, it's lacking definition and shading, and it contradicts the desc. If you're set on changing the sprite for all those ammo boxes, change the desc accordingly.

Glloyd commented on maps/torch/items/explo_shotgun.dm:
```diff
@@ -0,0 +1,91 @@
+					C.apply_damage(rand(10,20), def_zone=zone, blocked=C.run_armor_check(zone))
+			else
+				visible_message("<span class='danger'>[src] explodes!</span>")
+			explosion(get_turf(src), -1, -1, 1)
```
Is the explosion really necessary here? You're going to be doing double damage to the person holding it.

comma replied:
Very necessary

Glloyd replied:
Okay, but why both the damage and the explosion? Why not one or the other?

comma replied:
Explosion is cosmetic, it will barely do any damage, especially to someone wearing a voidsuit.  Handstuff is actual damage dealing here.

babydoll commented:
You got some `step_` in the maps it appears

comma committed:
Ammo boxes

Moves them under single parent so I don't have to copypaste shit everywhere.

comma committed:
Adds shotgun and ammo to supply packs

Shotgun is overpriced (same price as two REAL shotguns), ammo is same as pure beanbags crate.

comma committed:
Fixes DME

Whoops ticked in map specific file

babydoll reviewed (approved):
See how it goes

PigeonLord commented:
Guns that jam and spray like hell, 
With their stocks made by Mattel, 
Yeah, you'll always own the latest weaponry!
